monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -93,59 +107,59 @@ def connect(
             timeout=MEDIUM_REQUEST_TIMEOUT,
         )
         response.raise_for_status()
-
         self._api_url = f"https://{island_server}/api"
 
-    def _get(self, endpoint: str, timeout: float) -> requests.Response:
-        url = f"{self._api_url}/{endpoint}"
-        logger.debug(f"GET {url}, timeout={timeout}")
-
-        return self._session.get(url, verify=False, timeout=timeout)  # noqa: DUO123
-
-    def _post(self, endpoint: str, data: JSONSerializable, timeout: float) -> requests.Response:
+    def _send_request(
+        self,
+        request_type: RequestType,
+        endpoint: str,
+        timeout=MEDIUM_REQUEST_TIMEOUT,
+        data=None,
+        *args,
+        **kwargs,
+    ) -> requests.Response:
         url = f"{self._api_url}/{endpoint}"
-        logger.debug(f"POST {url}, timeout={timeout}")
-
-        return self._session.post(url, json=data, verify=False, timeout=timeout)  # noqa: DUO123
+        logger.debug(f"{request_type.name} {url}, timeout={timeout}")
 
-    def _put(self, endpoint: str, data: JSONSerializable, timeout: float) -> requests.Response:
-        url = f"{self._api_url}/{endpoint}"
-        logger.debug(f"PUT {url}, timeout={timeout}")
+        method = self.supported_requests[request_type]
+        response = method(url, *args, timeout=timeout, verify=False, json=data, **kwargs)
+        response.raise_for_status()
 
-        return self._session.put(url, json=data, verify=False, timeout=timeout)  # noqa: DUO123
+        return response
 
     @handle_island_errors
     def send_log(self, agent_id: AgentID, log_contents: str):
-        response = self._put(f"agent-logs/{agent_id}", log_contents, MEDIUM_REQUEST_TIMEOUT)
-        response.raise_for_status()
+        self._send_request(
```

Comment:
As a note, this should have been a different PR. The goal of this PR can be achieved without these changes, and now these changes will hold up the PR.

Keep things small and focused.

Reply:
I don't think that having to read an argument is not readable. The readability could be improved by adding keyword arguments, it will make it clearer what's happening. I'll look into how hard is it to generate the wrappers, but IMO this is already an improvement.